timeseries/src/autogluon/timeseries/trainer/abstract_trainer.py

```diff
@@ -572,10 +570,28 @@ def _train_multi(
                 )
 
         if self.enable_ensemble:
-            try:
-                model_names_trained.append(self.fit_ensemble(val_data=val_data, model_names=model_names_trained))
-            except Exception as e:  # noqa
-                logger.error(f"\tEnsemble training failed with error \n{traceback.format_exc()}.")
+            models_available_for_ensemble = self.get_model_names()
+
+            time_left_for_ensemble = None
+            if time_limit is not None:
+                time_left_for_ensemble = time_limit - (time.time() - time_start)
+
+            if time_left_for_ensemble is not None and time_left_for_ensemble <= 0:
+                logger.info(
+                    "Not fitting ensemble due to lack of time remaining. "
+                    "Time left: {time_left_for_ensemble:.2f} seconds"
+                )
+            elif len(models_available_for_ensemble) <= 1:
+                logger.info(f"Not fitting ensemble as only {len(models_available_for_ensemble)} models were trained.")
```

Comment:
Maybe have a separate error message for "all models have failed" (`len(models_available_for_ensble) == 0`) and "only 1 model was trained" (`len(models_available_for_ensble) == 1`)?